Core/Object Arts/Dolphin/Base/DateAndTime.cls

```diff
@@ -656,10 +656,21 @@ stbConvertFrom: anSTBClassFormat
 
 	^
 	[:data |
-	| totalSeconds offset |
-	offset := data second.
-	totalSeconds := data first - 59958230400 - offset asSeconds.
-	self utcSeconds: totalSeconds offset: offset asSeconds]!
+		(data at: 2) class == Duration ifTrue: [		"Original code satisfies one set of tests"
+			| totalSeconds offset |
+			offset := data second.
+			totalSeconds := data first - 59958230400 - offset asSeconds.
+			self utcSeconds: totalSeconds offset: offset asSeconds.
+		] ifFalse: [							"This satisfied another format found in the wild"
+			| days seconds offset totalSeconds |
+			days := data at: 1.		"Days since Sunday, December 31, 0000"
+			days := days - 693961.	"Days since December 31, 1900"
+			seconds := data at: 2.	"Likely to be a Fraction"
```

Comment:
These instances are not from Ian's original definition. They are from the window between PR's #741 and #748 - hence why an instance created with code from 24th April will be of this form, as that is in the window between the PRs.
They should not require any data conversion because the inst var values should be correct as is. Consequently the code can be simplified to:

```
	^
	[:data |
	| iv2 |
	iv2 := data second.
	iv2 class == Duration
		ifTrue: 
			[| totalSeconds |
			"Convert from IDB format DateAndTime"
			totalSeconds := data first - 59958230400 - iv2 asSeconds.
			self utcSeconds: totalSeconds offset: iv2 asSeconds]
		ifFalse: 
			["The object is already of the right format, but was saved after the addition of DateAndTime to the base but before the STB version upgrade for compatibility with the original IDB DateAndTimes was added."
			self basicNew
				setRataDie: data first
				seconds: iv2
				offset: data third]]
```